### DISCUSSION TITLE
I need help in the source code

### DISCUSSION THREAD

**Discussion by THEkatinamicrowave** [CodenameCrew/CodenameEngine#295] [Q&A] [answered]
I was editing code in the source because I can't be bothered to learn Codename stuff, and I came across some errors.

```powershell
source/funkin/menus/PeterCutsceneIntro.hx:9: characters 9-19 : Unknown identifier : startVideo
source/funkin/menus/BetaWarningState.hx:58: characters 20-44 : Not enough arguments, expected path:String, callback:() -> Void
source/funkin/menus/BetaWarningState.hx:58: characters 20-44 : ... For function argument 'nextState'
``` 

I'll paste any changes I've made to either scripts down here. Please give suggestions on how this could possibly make sucky errors happen!

BetaWarningState errors and changes:
```haxe
private function goToTitle() {
	MusicBeatState.skipTransIn = MusicBeatState.skipTransOut = true;
	FlxG.switchState(new PeterCutsceneIntro());
}
```

PeterCutsceneIntro.hx:

```haxe
package funkin.menus;

import funkin.game.cutscenes.VideoCutscene;
import funkin.game.cutscenes.ScriptedCutscene;

class PeterCutsceneIntro extends VideoCutscene {
    
    override function create() {
        startVideo(Paths.video("betergriffin"), function() {
            close;
        });
    }

}
```

**Comment by NexIsDumb** [comment, marked as answer]
okay first of all, for startVideo you gotta extend ScriptedCutscene and not VideoCutscene, so keep VideoCutscene and instead do:
```haxe
package funkin.menus;

import funkin.game.cutscenes.VideoCutscene;

class PeterCutsceneIntro extends VideoCutscene {
    
    public function new() {
        super(Paths.video("betergriffin"), close);
    }

}
```
Tell me if it worked cause I didnt test it and might have done some small errors

**Reply by THEkatinamicrowave** [reply]
Thank you yeah that was a stupid on my end. I think I'm getting the idea of Haxe code but yeah I've got a bit to go.
Anyway yeah the only errors left are the goToTitle() ones

**Reply by NexIsDumb** [reply]
wait, paste the error here

**Reply by THEkatinamicrowave** [reply]
```powershell
source/funkin/menus/BetaWarningState.hx:58: characters 20-44 : Not enough arguments, expected scriptPath:String, callback:() -> Void
source/funkin/menus/BetaWarningState.hx:58: characters 20-44 : ... For function argument 'nextState'
```
In the meantime, I'll try to transition from PeterCutsceneIntro.hx to TitleState.hx

**Reply by THEkatinamicrowave** [reply]
oh yeah and the part that's erroring is this in case i yeah

```haxe
private function goToTitle() {
		MusicBeatState.skipTransIn = MusicBeatState.skipTransOut = true;
		FlxG.switchState(new PeterCutsceneIntro());
	}
```

**Reply by NexIsDumb** [reply]
> oh yeah and the part that's erroring is this in case i yeah
> 
> 
> 
> ```haxe
> 
> private function goToTitle() {
> 
> 		MusicBeatState.skipTransIn = MusicBeatState.skipTransOut = true;
> 
> 		FlxG.switchState(new PeterCutsceneIntro());
> 
> 	}
> 
> ```

wait even after placing what I did?? huh

**Reply by THEkatinamicrowave** [reply]
yeah I made a separate issue for a different problem
the code you made worked sorry if that wasn't clear !

**Reply by NexIsDumb** [reply]
ohh okay!
